server/src/main/java/org/apache/druid/segment/realtime/appenderator/UnifiedIndexerAppenderatorsManager.java

```diff
@@ -196,4 +236,288 @@ public boolean shouldTaskMakeNodeAnnouncements()
   {
     return false;
   }
+
+  @Override
+  public void shutdown()
+  {
+    if (mergeExecutor != null) {
+      mergeExecutor.shutdownNow();
```

Comment:
After shutting down the `Executor`, we should explicitly set it to `null` which is more friendly to the GC.

Reply:
Changed this to set to null after shutdown